listenbrainz/webserver/static/js/src/listens/ListenCard.tsx

```diff
@@ -326,12 +329,24 @@ export default class ListenCard extends React.Component<
             <div className="listen-details">{listenDetails}</div>
           ) : (
             <div className="listen-details">
-              <div title={trackName} className="ellipsis-2-lines">
-                {getTrackLink(listen)}
-              </div>
-              <span className="small text-muted ellipsis" title={artistName}>
+              {trackDuration ? (
+                <div className="title-duration">
+                  <div title={trackName} className="ellipsis-2-lines">
+                    {getTrackLink(listen)}
+                  </div>
+                  <div className="small text-muted" title="Duration">
+                    {isNumber(trackDuration) &&
+                      millisecondsToStr(trackDuration)}
+                  </div>
+                </div>
+              ) : (
+                <div title={trackName} className="ellipsis-2-lines">
+                  {getTrackLink(listen)}
+                </div>
+              )}
```

Comment:
I would suggest always rendering the title-duration parent element, and only conditionally render the duration element:

```suggestion
                <div className="title-duration">
                  <div title={trackName} className="ellipsis-2-lines">
                    {getTrackLink(listen)}
                  </div>
                  {trackDuration ? (
                    <div className="small text-muted" title="Duration">
                      {isNumber(trackDuration) &&
                        millisecondsToStr(trackDuration)}
                    </div>
                    )}
                </div>
```

Reply:
I'll test this and let you know :)

Reply:
I couldn't commit this suggestion because the usage of the ternary operator is incorrect (missing `:`). But because of that I was able to test it locally. I have included it in the latest commit.